src/Discord.Net.SlashCommands/Info/ModuleInfo.cs

```diff
@@ -0,0 +1,206 @@
+using Discord.SlashCommands.Builders;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Discord.SlashCommands
+{
+    /// <summary>
+    /// Contains the information of a Slash command module
+    /// </summary>
+    public class ModuleInfo
+    {
+        internal ILookup<string, PreconditionAttribute> _groupedPreconditions { get; }
+
+        /// <summary>
+        /// Command service this module belongs to
+        /// </summary>
+        public SlashCommandService CommandService { get; }
```

Comment:
Dont know if a back reference is a good idea for this. Don't see a need for a backreference

Reply:
It is need to access the configuration `Commands` modules stored the command service reference